src/_P064_APDS9960.ino

```diff
@@ -102,25 +199,32 @@ boolean Plugin_064(byte function, struct EventStruct *event, String& string)
         PLUGIN_064_pds = new SparkFun_APDS9960();
 
         String log = F("APDS : ");
-        if ( PLUGIN_064_pds->init() )
+        if (PLUGIN_064_pds->init(PCONFIG(2), PCONFIG(3), PCONFIG(5), PCONFIG(6), PCONFIG(7)) )
         {
           log += F("Init");
 
           PLUGIN_064_pds->enablePower();
 
           if (! PLUGIN_064_pds->enableLightSensor(false))
             log += F(" - Error during light sensor init!");
-          if (! PLUGIN_064_pds->enableProximitySensor(false))
-            log += F(" - Error during proximity sensor init!");
+          if (PCONFIG(1) == PLUGIN_MODE_GPL_064) { // Gesture/Proximity/ALS mode
+            if (! PLUGIN_064_pds->enableProximitySensor(false))
+              log += F(" - Error during proximity sensor init!");
 
-          if (! PLUGIN_064_pds->enableGestureSensor(false))
-            log += F(" - Error during gesture sensor init!");
+            if (! PLUGIN_064_pds->enableGestureSensor(false, PCONFIG(4)))
+              log += F(" - Error during gesture sensor init!");
+          }
         }
         else
         {
           log += F("Error during APDS-9960 init!");
         }
 
+        // Forced reset values
+        UserVar[event->BaseVarIndex + 0] = 0.0;
+        UserVar[event->BaseVarIndex + 1] = 0.0;
+        UserVar[event->BaseVarIndex + 2] = 0.0;

```

Comment:
You reset the values at init, but this is also called after a warm boot.
So there may be values restored from RTC. Is this desired behavior on a warm boot?

Reply:
I moved that to where the values names are swapped, as the old values are unrelated to any new ones.